File: frontend/src/types/product.ts
export interface ProductOptionValue {
    id: number;
    value: string;
}

export interface ProductOption {
    id: number;
    name: string;
    values: ProductOptionValue[];
}

export interface ProductVariant {
    id: number;
    sku: string;
    basePrice: number;
    salePrice?: number;
    costPrice?: number;
    stockQuantity: number;
    images: string[]; // Changed from single image to array
    optionValues: ProductOptionValue[];
}

export interface Product {
    id: number;
    name: string;
    slug: string;
    description?: string;
    stockQuantity: number;
    sku?: string;
    category?: Category;
    categoryId?: number;
    images: string[];
    status: 'ACTIVE' | 'DRAFT' | 'ARCHIVED';
    featured: boolean;
    viewCount: number;
    soldCount: number;
    createdAt: string;
    updatedAt: string;
    hasVariants: boolean;
    options: ProductOption[];
    variants: ProductVariant[];

    // New Fields
    basePrice?: number;
    salePrice?: number;
    costPrice?: number;
    brand?: string;
    weight?: number;
    length?: number;
    width?: number;
    height?: number;
    metaTitle?: string;
    metaDescription?: string;
    metaKeywords?: string;
    averageRating?: number;
    reviewCount?: number;
    totalSales?: number;

    isDiscontinued?: boolean;
}

export interface Category {
    id: number;
    name: string;
    slug: string;
    description?: string;
    imageUrl?: string;
    parentId?: number;
    status: 'ACTIVE' | 'DRAFT' | 'ARCHIVED';
    displayOrder: number;
    children?: Category[];
}

File: backend/src/main/java/com/ecommerce/model/entity/Product.java
package com.ecommerce.model.entity;

import com.ecommerce.model.enums.ProductStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "products")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(unique = true, length = 200)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    @Builder.Default
    private Integer stockQuantity = 0;

    @Column(length = 100)
    private String sku;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ElementCollection
    @CollectionTable(name = "product_images", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "image_url", length = 500)
    @Builder.Default
    private List<String> images = new ArrayList<>();

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private ProductStatus status = ProductStatus.ACTIVE;

    @Column(nullable = false)
    @Builder.Default
    private Boolean featured = false;

    @Column(nullable = false)
    @Builder.Default
    private Integer viewCount = 0;

    @Column(nullable = false)
    @Builder.Default
    private Integer soldCount = 0;

    @Column(nullable = false)
    @Builder.Default
    private Boolean hasVariants = false;

    // Pricing
    @Column(precision = 10, scale = 2)
    private BigDecimal basePrice;

    @Column(precision = 10, scale = 2)
    private BigDecimal salePrice;

    @Column(precision = 10, scale = 2)
    private BigDecimal costPrice;

    // Attributes
    @Column(length = 100)
    private String brand;

    private Double weight;
    private Double length;
    private Double width;
    private Double height;

    // SEO
    @Column(length = 200)
    private String metaTitle;

    @Column(columnDefinition = "TEXT")
    private String metaDescription;

    private String metaKeywords;

    // Stats
    @Builder.Default
    private Double averageRating = 0.0;

    @Builder.Default
    private Integer reviewCount = 0;

    @Builder.Default
    private Integer totalSales = 0;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<ProductOption> options = new ArrayList<>();

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<ProductVariant> variants = new ArrayList<>();

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<ProductTranslation> translations = new ArrayList<>();

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;
}
